webapp/packages/plugin-data-viewer/src/DatabaseDataModel/DatabaseDataManager.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { makeObservable, observable } from 'mobx';
+
+import type { GraphQLService } from '@cloudbeaver/core-sdk';
+import { download, GlobalConstants } from '@cloudbeaver/core-utils';
+
+import type { IResultSetElementKey } from './Actions/ResultSet/IResultSetDataKey';
+import { isResultSetContentValue } from './Actions/ResultSet/ResultSetContentValue';
+import { ResultSetDataAction } from './Actions/ResultSet/ResultSetDataAction';
+import { ResultSetDataElementUtils } from './Actions/ResultSet/ResultSetDataElementUtils';
+import type { IResultSetValue } from './Actions/ResultSet/ResultSetFormatAction';
+import { ResultSetViewAction } from './Actions/ResultSet/ResultSetViewAction';
+import type { IDatabaseDataManager } from './IDatabaseDataManager';
+import type { IDatabaseDataSource } from './IDatabaseDataSource';
+import type { IDatabaseResultSet } from './IDatabaseResultSet';
+
+const RESULT_VALUE_PATH = 'sql-result-value';
+
+export class DatabaseDataManager<TSource extends IDatabaseDataSource<any, any>> implements IDatabaseDataManager {
+  private readonly cache: Map<string, string>;
+  activeElement: IResultSetElementKey | null;
+
+  constructor(
+    private readonly graphQLService: GraphQLService,
+    private readonly source: TSource
+  ) {
+    this.cache = new Map();
+    this.activeElement = null;
+
+    makeObservable<DatabaseDataManager<TSource>, 'cache'>(this, {
+      cache: observable,
+      activeElement: observable.ref,
+    });
+  }
+
+  canGetFileURLFor(element: IResultSetElementKey, resultIndex: number) {
+    const view = this.source.getAction(resultIndex, ResultSetViewAction);
+    const cellValue = view.getCellValue(element);
+
+    return isResultSetContentValue(cellValue);
+  }
+
+  async getFileURLFor(element: IResultSetElementKey, resultIndex: number) {
+    const result = this.source.getResult(resultIndex);
+    const data = this.source.getAction(resultIndex, ResultSetDataAction);
+    const column = data.getColumn(element.column);
+    const row = data.getRowValue(element.row);
+
+    if (!result || !row || !column) {
+      throw new Error('Failed to get arguments data');
```

Comment:
the user knows nothing about `arguments` please write a more detailed explanation, like: `Failed to get value metadata information`